parl/algorithms/torch/sac.py

```diff
@@ -0,0 +1,134 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import parl
+import torch
+from torch.distributions import Normal
+import torch.nn.functional as F
+from copy import deepcopy
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+__all__ = ['SAC']
+
+
+class SAC(parl.Algorithm):
+    def __init__(self,
+                 model,
+                 max_action,
```

Comment:
why should we need to pass `max_action` while we have `action_space` in parameter list ?

Reply:
The action must be distributed between action_space.low[0] and action_space.high[0], which are opposite numbers(generally it's the case in gym envs).
So in this case, we can let 
```
max_action = action_space.high[0]
action =  torch.tanh(input) * max_action
```